File: backend/domains/task/http/dto/task.go
package dto

// CreateTaskRequest 创建任务请求
type CreateTaskRequest struct {
	Title       string   `json:"title" binding:"required,min=1,max=200"`
	Description string   `json:"description" binding:"max=5000"`
	Priority    string   `json:"priority" binding:"omitempty,oneof=low medium high"`
	DueDate     string   `json:"due_date" binding:"omitempty,datetime=2006-01-02T15:04:05Z07:00"`
	Tags        []string `json:"tags" binding:"omitempty,max=10,dive,max=50"`
}

// CreateTaskResponse 创建任务响应
type CreateTaskResponse struct {
	TaskID    string `json:"task_id"`
	Title     string `json:"title"`
	Status    string `json:"status"`
	CreatedAt string `json:"created_at"`
}

// UpdateTaskRequest 更新任务请求
type UpdateTaskRequest struct {
	Title       string   `json:"title" binding:"omitempty,min=1,max=200"`
	Description string   `json:"description" binding:"max=5000"`
	Priority    string   `json:"priority" binding:"omitempty,oneof=low medium high"`
	DueDate     string   `json:"due_date" binding:"omitempty,datetime=2006-01-02T15:04:05Z07:00"`
	Tags        []string `json:"tags" binding:"omitempty,max=10,dive,max=50"`
}

// UpdateTaskResponse 更新任务响应
type UpdateTaskResponse struct {
	TaskID    string `json:"task_id"`
	Title     string `json:"title"`
	Status    string `json:"status"`
	UpdatedAt string `json:"updated_at"`
}

// CompleteTaskResponse 完成任务响应
type CompleteTaskResponse struct {
	TaskID      string `json:"task_id"`
	Status      string `json:"status"`
	CompletedAt string `json:"completed_at"`
}

// DeleteTaskResponse 删除任务响应
type DeleteTaskResponse struct {
	Success   bool   `json:"success"`
	DeletedAt string `json:"deleted_at"`
}

// GetTaskResponse 获取任务响应
type GetTaskResponse struct {
	TaskID      string   `json:"task_id"`
	Title       string   `json:"title"`
	Description string   `json:"description"`
	Status      string   `json:"status"`
	Priority    string   `json:"priority"`
	DueDate     *string  `json:"due_date"`
	Tags        []string `json:"tags"`
	CreatedAt   string   `json:"created_at"`
	UpdatedAt   string   `json:"updated_at"`
	CompletedAt *string  `json:"completed_at"`
}

// ListTasksRequest 列出任务请求
type ListTasksRequest struct {
	// 筛选参数
	Status      string `form:"status" binding:"omitempty,oneof=pending in_progress completed"`
	Priority    string `form:"priority" binding:"omitempty,oneof=low medium high"`
	Tag         string `form:"tag"`
	DueDateFrom string `form:"due_date_from" binding:"omitempty,datetime=2006-01-02"`
	DueDateTo   string `form:"due_date_to" binding:"omitempty,datetime=2006-01-02"`
	Keyword     string `form:"keyword" binding:"omitempty,max=100"`

	// 排序参数
	SortBy    string `form:"sort_by" binding:"omitempty,oneof=created_at due_date priority"`
	SortOrder string `form:"sort_order" binding:"omitempty,oneof=asc desc"`

	// 分页参数
	Page  int `form:"page" binding:"omitempty,min=1"`
	Limit int `form:"limit" binding:"omitempty,min=1,max=100"`
}

// TaskItem 任务列表项
type TaskItem struct {
	TaskID    string   `json:"task_id"`
	Title     string   `json:"title"`
	Status    string   `json:"status"`
	Priority  string   `json:"priority"`
	DueDate   *string  `json:"due_date"`
	Tags      []string `json:"tags"`
	CreatedAt string   `json:"created_at"`
}

// ListTasksResponse 列出任务响应
type ListTasksResponse struct {
	Tasks      []TaskItem `json:"tasks"`
	TotalCount int        `json:"total_count"`
	Page       int        `json:"page"`
	Limit      int        `json:"limit"`
	HasMore    bool       `json:"has_more"`
}

// ErrorResponse 错误响应
type ErrorResponse struct {
	Error   string `json:"error"`             // 错误码
	Message string `json:"message"`           // 错误消息
	Details string `json:"details,omitempty"` // 详细信息（可选）
}

File: backend-nodejs/domains/task/http/dto/task.ts
/**
 * Task HTTP DTO
 * 定义 HTTP 请求和响应的数据结构
 */

import { z } from 'zod';

// ========================================
// CreateTask
// ========================================

export const CreateTaskRequestSchema = z.object({
  title: z.string().min(1, '任务标题不能为空').max(200, '任务标题过长，最大 200 字符'),
  description: z.string().max(5000, '描述过长，最大 5000 字符').optional(),
  priority: z.enum(['low', 'medium', 'high']).optional(),
  due_date: z.string().datetime().optional(),
  tags: z.array(z.string()).max(10, '标签过多，最多 10 个').optional(),
});

export type CreateTaskRequest = z.infer<typeof CreateTaskRequestSchema>;

export interface CreateTaskResponse {
  task_id: string;
  title: string;
  status: string;
  created_at: string; // ISO 8601
}

// ========================================
// UpdateTask
// ========================================

export const UpdateTaskRequestSchema = z.object({
  title: z.string().min(1).max(200).optional(),
  description: z.string().max(5000).optional(),
  priority: z.enum(['low', 'medium', 'high']).optional(),
  due_date: z.string().datetime().optional(),
  tags: z.array(z.string()).max(10).optional(),
});

export type UpdateTaskRequest = z.infer<typeof UpdateTaskRequestSchema>;

export interface UpdateTaskResponse {
  task_id: string;
  title: string;
  status: string;
  updated_at: string; // ISO 8601
}

// ========================================
// Params Schemas (for route params)
// ========================================

export const TaskParamsSchema = z.object({
  id: z.string().min(1, '任务 ID 不能为空'),
});

export type TaskParams = z.infer<typeof TaskParamsSchema>;

// ========================================
// CompleteTask
// ========================================

export interface CompleteTaskResponse {
  task_id: string;
  status: string;
  completed_at: string; // ISO 8601
}

// ========================================
// DeleteTask
// ========================================

export interface DeleteTaskResponse {
  success: boolean;
  deleted_at: string; // ISO 8601
}

// ========================================
// GetTask
// ========================================

export interface GetTaskResponse {
  task_id: string;
  title: string;
  description: string;
  status: string;
  priority: string;
  due_date?: string; // ISO 8601
  tags: string[];
  created_at: string; // ISO 8601
  updated_at: string; // ISO 8601
  completed_at?: string; // ISO 8601
}

// ========================================
// ListTasks
// ========================================

export const ListTasksQuerySchema = z.object({
  status: z.enum(['pending', 'in_progress', 'completed']).optional(),
  priority: z.enum(['low', 'medium', 'high']).optional(),
  tag: z.string().optional(),
  due_date_from: z.string().datetime().optional(),
  due_date_to: z.string().datetime().optional(),
  keyword: z.string().optional(),
  sort_by: z.enum(['created_at', 'due_date', 'priority']).optional(),
  sort_order: z.enum(['asc', 'desc']).optional(),
  page: z.coerce.number().int().positive().default(1).optional(),
  limit: z.coerce.number().int().positive().max(100).default(20).optional(),
});

export type ListTasksQuery = z.infer<typeof ListTasksQuerySchema>;

export interface TaskItem {
  task_id: string;
  title: string;
  status: string;
  priority: string;
  due_date?: string; // ISO 8601
  tags: string[];
  created_at: string; // ISO 8601
}

export interface ListTasksResponse {
  tasks: TaskItem[];
  total_count: number;
  page: number;
  limit: number;
  has_more: boolean;
}

// ========================================
// Error Response
// ========================================

export interface ErrorResponse {
  error: string; // 错误码
  message: string; // 错误消息
  details?: string; // 详细信息（可选）
}
